### ISSUE TITLE
meta(invariants): tracking issue for improvements to invariant testing

### ISSUE THREAD

**Issue by mds1** [foundry-rs/foundry#4438]
### Component

Forge

### Describe the feature you would like

WIP

**Comment by zerosnacks** [comment]
Noticing a bit of flakyness with invariant tests recently, for example see: https://github.com/foundry-rs/foundry/actions/runs/10298801797/job/28504775878 / https://github.com/foundry-rs/foundry/actions/runs/10286582825/job/28467603861

cc @grandizzy

**Comment by grandizzy** [comment]
> Noticing a bit of flakyness with invariant tests recently in the CI, for example see: https://github.com/foundry-rs/foundry/actions/runs/10298801797/job/28504775878 / https://github.com/foundry-rs/foundry/actions/runs/10286582825/job/28467603861
> 
> cc @grandizzy

yep, this should make it faster and avoid flakyness :crossed_fingers:  https://github.com/foundry-rs/foundry/pull/8634
